### PR TITLE
Allow forcemerge in the hot phase for ILM policies

### LINKED ISSUE #43165: Add Force Merge to hot lifecycle phase
**Add Force Merge to hot lifecycle phase**:
For time series data it makes sense that you may want to do a force merge on an index in the hot phase after rollover occurs. Seems pretty straight forward but if you want more information LMK.

### PR DESCRIPTION
**Pull request by dakrone**
This commit changes the `forcemerge` action to also be allowed in the `hot` phase for policies. The
forcemerge will occur after a rollover, and allows users to take advantage of higher disk speeds for
performing the force merge (on a separate node type, for example).

On caveat with this is that a `forcemerge` in the `hot` phase *MUST* be accompanied by a `rollover`
action. ILM validates policies to ensure this is the case.

Resolves #43165

### CODE REVIEW

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/LifecyclePolicyTests.java [hunk 1]
```diff
@@ -145,6 +146,24 @@ public static LifecyclePolicy randomTimeseriesLifecyclePolicyWithAllPhases(@Null
     }
 
     public static LifecyclePolicy randomTimeseriesLifecyclePolicy(@Nullable String lifecycleName) {
+        LifecyclePolicy policy = null;
+
+        // Keep generating a random policy until it passes the validation,
+        // ignoring known specific invalid states
+        while (policy == null) {
```
Comment (andreidan) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/LifecyclePolicyTests.java]: Would it be a bit faster to validate when generating the random policy that if we added `forcemerge` we will add `rollover`?

**Review by andreidan — APPROVED** [review]
Thanks for adding support for this Lee, nicely done! 

I have a possible optimisation suggestion to keep the tests running time to a minimum, but up to you if you'd like to change it (these are unit tests so they're blazing fast anyway)

LGTM

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/TimeseriesLifecycleType.java [hunk 2]
```diff
@@ -195,5 +200,21 @@ public void validate(Collection<Phase> phases) {
...
+            // That contains the 'forcemerge' action
+            .filter(phase -> phase.getActions().containsKey(ForceMergeAction.NAME))
+            // But does *not* contain the 'rollover' action?
+            .filter(phase -> phase.getActions().containsKey(RolloverAction.NAME) == false)
+            // If there is, throw an exception
+            .findAny()
+            .ifPresent(phase -> {
+                throw new IllegalArgumentException("the [" + ForceMergeAction.NAME +
+                    "] action may not be used in the [" + phase.getName() +
+                    "] phase without an accompanying [" + RolloverAction.NAME + "] action");
```
Comment (probakowski) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/TimeseriesLifecycleType.java]: Can be simplified slightly:
```java
if (phases.stream()
    // Is there a hot phase
    .filter(phase -> HOT_PHASE.equals(phase.getName()))
    // That contains the 'forcemerge' action
    .filter(phase -> phase.getActions().containsKey(ForceMergeAction.NAME))
    // But does *not* contain the 'rollover' action?
    .anyMatch(phase -> phase.getActions().containsKey(RolloverAction.NAME) == false)) {
    // If there is, throw an exception
    throw new IllegalArgumentException("the [" + ForceMergeAction.NAME +
        "] action may not be used in the [" + HOT_PHASE +
        "] phase without an accompanying [" + RolloverAction.NAME + "] action");
}
```

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/LifecyclePolicyTests.java (hunk 1 above)
Comment (probakowski) [reply x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/LifecyclePolicyTests.java]: +1 for @andreidan idea, currently we risk potentially very long running tests (or random timeouts) if we can't generate correct policy

**Review by probakowski — APPROVED** [review]
LGTM, thanks @dakrone ! I've added one small nit (optional)

**Pull request merged**